akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala

```diff
@@ -723,4 +730,73 @@ final class ProcessorModulePhase(materializer: PhasedFusingActorMaterializer) ex
   override def takePublisher(slot: Int, publisher: Publisher[Any]): Unit = publisher.subscribe(processor)
 
   override def onIslandReady(): Unit = ()
+}
+
+object TlsModuleIslandTag extends IslandTag
+
+final class TlsModulePhase(materializer: PhasedFusingActorMaterializer) extends PhaseIsland[TlsModulePhase.MaterializedInfo] {
+  import TlsModulePhase._
+
+  def name: String = "TlsModulePhase"
+
+  def materializeAtomic(mod: AtomicModule[Shape, Any], attributes: Attributes): (MaterializedInfo, Any) = {
+    val es = materializer.effectiveSettings(attributes)
+    println(mod)
+    val tls = mod.asInstanceOf[TlsModule]
+
+    val props = TLSActor.props(materializer.settings, tls.createSSLEngine, tls.verifySession, tls.closing)
+    val impl = materializer.actorOf(props, s"tls-${nextId()}", es.dispatcher)
```

Comment:
I think the ID generation should be moved into the `actorOf` method instead. This is related to what @patriknw is working on, so maybe coordinate on this?